### PR TITLE
Add description metadata to Grafana Agent docs

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

This PR adds the description metatdata to the entire Agent doc set. This is foundation work to fix the build validation errors in the documentation.

The `tools/gen-crd-docs/template/pkg.tpl` is also updated to add the description metadata to the generated api.md

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Review by jdbaldry — APPROVED** [review]
Nice!

File: docs/sources/flow/reference/cli/fmt.md [hunk 1]
```diff
@@ -6,7 +6,8 @@ aliases:
 canonical: https://grafana.com/docs/agent/latest/flow/reference/cli/fmt/
 description: The `fmt` command formats a Grafana Agent configuration file.
 menuTitle: fmt
-title: fmt command
+title: `fmt`` command
```
Comment (jdbaldry) [inline docs/sources/flow/reference/cli/fmt.md:9]:
```suggestion
title: fmt command
```

You can't use Markdown in the titles

Comment (jdbaldry) [reply docs/sources/flow/reference/cli/fmt.md:9]: Actually that suggestion looks incorrect given the convention you've adopted with the other commands of "The `<NAME>` command"

Comment (clayton-cornell) [reply docs/sources/flow/reference/cli/fmt.md]: Yeah this is one I fixed local and forgot to push it yesterday :-P It was late.  Fixed to match the other CLI topic titles.

**Review by mattdurham — APPROVED** [review]
I looked at the top 10 files, surely thats good enough.

**Comment by clayton-cornell** [comment]
Not backporting this metadata change to v0.34. The delta is too big for this change, and it's not worth the extra time it'll take to fix the backport.

**Pull request merged**